tests/common/exceptions.h

```diff
@@ -0,0 +1,302 @@
+#include <string>
+#include <utility>
+
+CATCH_REGISTER_ENUM(sycl::errc, sycl::errc::success, sycl::errc::runtime,
+                    sycl::errc::kernel, sycl::errc::accessor,
+                    sycl::errc::nd_range, sycl::errc::event,
+                    sycl::errc::kernel_argument, sycl::errc::build,
+                    sycl::errc::invalid, sycl::errc::memory_allocation,
+                    sycl::errc::platform, sycl::errc::profiling,
+                    sycl::errc::feature_not_supported,
+                    sycl::errc::kernel_not_supported,
+                    sycl::errc::backend_mismatch);
+
+namespace sycl_cts {
+namespace detail {
+namespace has_member {
+
+// Safely ensures class member availability w/o compile-time failures using
+// SFINAE
+template <typename, typename = void>
+struct code : std::false_type {};
+template <typename T>
+struct code<T, std::void_t<decltype(std::declval<T>().code())>>
+    : std::true_type {};
+
+template <typename, typename = void>
+struct category : std::false_type {};
+template <typename T>
+struct category<T, std::void_t<decltype(std::declval<T>().category())>>
+    : std::true_type {};
+
+}  // namespace has_member
+
+namespace has_function {
+
+// Safely ensures free function availability w/o compile-time failures
+template <typename = void>
+struct sycl_make_error_code : std::false_type {};
+template <>
+struct sycl_make_error_code<
+    std::void_t<decltype(sycl::make_error_code(std::declval<sycl::errc>()))>>
+    : std::true_type {};
+
+}  // namespace has_function
+}  // namespace detail
+
+/**
+ * Helper function to get details for SYCL exception
+ * Can be used, for example, for INFO(... << stringify_sycl_exception(ex))
+ * expressions
+ *
+ * TODO: Move function body to the common/exceptions.cpp
+ */
+inline std::string stringify_sycl_exception(const sycl::exception& e) {
+  std::ostringstream out;
+  out << "SYCL exception\n";
+
+  // Define helpers to format exception details
+  auto append_str = [&out](const char* description, std::string&& value) {
+    // Uses r-value reference directly
+    out << "with " << description << ": '" << value << "'\n";
+  };
+  auto append_cstr = [&append_str](const char* description, const char* value) {
+    if (!value) {
+      value = "nullptr";
+    }
+    append_str(description, {value});
+  };
+
+  // Collect exception details, considering possible implementation gaps
+  if constexpr (!detail::has_member::category<sycl::exception>::value) {
+    append_cstr("category", "not supported by implementation");
+  } else {
+    // Using reference to avoid object slicing
+    const auto& category = e.category();
+    append_cstr("category name", category.name());
+  }
+
+  if constexpr (!detail::has_member::code<sycl::exception>::value) {
+    append_cstr("code", "not supported by implementation");
+  } else {
+    // Using reference to avoid object slicing
+    const auto& code = e.code();
+    append_str("code value", std::to_string(code.value()));
```

Comment:
I think it would be very helpful to also print the corresponding `errc` enum value here!

Reply:
Great catch, thanks. Done by 94dfef0afb0b95d78bc093b163b197322f408de7